rtv/terminal.py

```diff
@@ -145,6 +155,40 @@ def suspend():
         finally:
             curses.doupdate()
 
+    @staticmethod
+    def select_program(file_url):
+        """
+        If there is a valid program to open the link, will return it, otherwise
+        returns None and the link will be opened with browser.
+        """
+        mimetype = mimetypes.guess_type(file_url)[0]
+        d = mailcap.getcaps()
+        try:
+            m = mailcap.findmatch(d, mimetype)
+        except AttributeError:
+            return None
+        candidates = []
+        if m[0] == None:
```

Comment:
In general I'm not quite following this logic.

- convention is to use ``m[0] is None`` instead of ``m[0] == None``
- If m[0] is not None, i.e. if mailcap finds a match, candidates is an empty list and the function returns None?
- Why search for matches manually, how is that different than what mailcap.findmatch() is already doing? 
- Adding d[key][0].items() and d[key][0].keys() looks like a typo. 

For example,
  ```
  >>> d['image/png'][0]
  {'description': 'PNG Image',
  'nametemplate': '%s.png',
  'test': 'test -n "$DISPLAY"',
  'view': "feh '%s'"}
  ```
  After you append to the list, you get
  ```
  [('test', 'test -n "$DISPLAY"'),
   ('nametemplate', '%s.png'),
   ('description', 'PNG Image'),
   ('view', "feh '%s'"),
   'test',
   'nametemplate',
   'description',
   'view']
  ```
This makes candidates look like a jumbled mess of strings when you do ``list(itertools.chain(*candidates))``
- What is the purpose of using the xdg library at all? I'm not at all familiar with how it works, but it's not clear to be what information is being gained.